zazen/ray_system.cc

```diff
@@ -0,0 +1,131 @@
+#include "ray_system.h"
+
+#include <float.h>
+#include <libzazen.h>
+
+#include "matrices.h"
+#include "vectors.h"
+
+static float RayOBBIntersection(Vector3 ray_origin, Vector3 ray_direction,
+                                Vector3 aabb_min, Vector3 aabb_max,
+                                float model_matrix[16]);
+
+void RaySystem::CalculateInterection(
+    struct zazen_ray_back_state *ray_back_state,
+    ZServer::CuboidWindowIterator *cuboid_window_iterator)
+{
+  struct zazen_cuboid_window_back_state *cuboid_window_state;
+  struct zazen_cuboid_window *focus_cuboid_window = nullptr;
+  float min_distance = FLT_MAX;
+
+  Vector3 ray_origin(ray_back_state->origin[0], ray_back_state->origin[1],
+                     ray_back_state->origin[2]);
+  Vector3 ray_direction(ray_back_state->direction[0],
+                        ray_back_state->direction[1],
+                        ray_back_state->direction[2]);
+
+  while ((cuboid_window_state = cuboid_window_iterator->Next())) {
+    float w = cuboid_window_state->width / 2;
+    float h = cuboid_window_state->height / 2;
+    float d = cuboid_window_state->depth / 2;
+    Vector3 aabb_min(-w, -h, -d);
+    Vector3 aabb_max(w, h, d);
+    float distance =
+        RayOBBIntersection(ray_origin, ray_direction, aabb_min, aabb_max,
+                           cuboid_window_state->model_matrix);
+    if (distance >= 0 && distance < min_distance) {
+      min_distance = d;
```

Comment:
```suggestion
      min_distance = distance;
```
実行して確認したわけじゃないけどここdistanceではない？

Reply:
その通りです。🙏